lib/ts/recipe/session/recipeImplementation.ts

```diff
@@ -375,43 +396,69 @@ export default function getRecipeInterface(
             return SessionFunctions.getSessionInformation(helpers, sessionHandle);
         },
 
+        /*
+            In all cases: if sIdRefreshToken token exists (so it's a legacy session) we clear it.
+            Check http://localhost:3002/docs/contribute/decisions/session/0008 for further details and a table of expected behaviours
+         */
         refreshSession: async function (
             this: RecipeInterface,
             { req, res, userContext }: { req: BaseRequest; res: BaseResponse; userContext: any }
         ): Promise<Session> {
             logDebugMessage("refreshSession: Started");
-            // We use a fallback mechanism here, to ensure there is a smooth upgrade path when switching transfer methods
-            // We only use it here and not while getting/validating sessions, because we want to "force" clients to upgrade
-            const outputTransferMethod = config.getTokenTransferMethod({ req, userContext });
-            let inputTransferMethod = outputTransferMethod;
-
-            let inputRefreshToken = getToken(config, req, "refresh", userContext, inputTransferMethod);
-            if (inputRefreshToken === undefined) {
-                inputTransferMethod = inputTransferMethod === "cookie" ? "header" : "cookie";
-                inputRefreshToken = getToken(config, req, "refresh", userContext, inputTransferMethod);
+
+            // This token isn't handled by getToken/setToken to limit the scope of this legacy/migration code
+            if (req.getCookieValue(LEGACY_ID_REFRESH_TOKEN_COOKIE_NAME) !== undefined) {
+                setCookie(config, res, LEGACY_ID_REFRESH_TOKEN_COOKIE_NAME, "", 0, "accessTokenPath");
             }
 
-            try {
-                if (inputRefreshToken === undefined) {
-                    logDebugMessage("refreshSession: UNAUTHORISED because refresh token from cookies is undefined");
-                    throw new STError({
-                        message:
-                            "Refresh token not found. Are you sending the refresh token in the request as a cookie?",
-                        type: STError.UNAUTHORISED,
-                    });
+            const refreshTokens: {
+                [key in TokenTransferMethod]?: string;
+            } = {};
+
+            // We check all token transfer methods for available refresh tokens
+            // We do this so that we can later clear all we are not overwrite
```

Comment:
"We do this so that we can later clear all we are not overwrite" -> what does this sentence mean?